### ISSUE TITLE
Fix `shorthand-property-no-redundant-values` false negatives for `var()`

### ISSUE THREAD

**Issue by gregmatys** [stylelint/stylelint#7656]
### What minimal example or steps are needed to reproduce the bug?

```css
a { padding: 0 var(--padding) 0; }
```

### What minimal configuration is needed to reproduce the bug?

```json
{ "rules": { "shorthand-property-no-redundant-values": true } }
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcCBOEIBcAEwA6A7baAtIQA4CGUUAlrgObwYBMADCQB4DceAvnnmVj5pylGvQzMMANzJIAFMRHU6ASgldcvXCAA0IAGZUANjAByZALZxEMNpZImAdGADOL3eAi5DtBCBy4GBjYILZoMLhQLiEMANp4QUEhLmgAniZGNESQ3lS0hClkkbJQIQkYALo8Hjk+AGIQSBZkaH4AVi5eHrAk7ogBSSAp6TCZuGgxwSBGLTApITrlyWkZWYS1eQVoRVAlkyEz4fMg1dxAA)

### Which Stylelint-related dependencies are you using?

```json { "stylelint": "15.6.0" } ```

### What did you expect to happen?

fix redundant values

### What actually happened?

nothing

### Do you have a proposal to fix the bug?

--

**Comment by ybiquitous** [comment]
@gregmatys Thanks for the report with the reproducible demo. This is a bug.

I've labeled the issue as **ready to implement**. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by ybiquitous** [comment]
"without eval" makes sense to me.

**Comment by ybiquitous** [comment]
I just opened PR #7657. It supports not only `var()`, but also other functions such as `calc()`. Please feel free to comment on the PR if you have any concerns. 🙏🏼

**Comment by romainmenke** [comment]
🤔 Is it an issue that with `var()` we can not know what that actual value will be?


```css
:root {
	--padding: 20px;
}

@media screen {
	:root {
		--padding: 20px 5px;
	}
}

.a {
	padding: 0 var(--padding) 0;
}

.b {
	padding: 0 20px 5px 0; /* no error */
}

.c {
	padding: 0 20px 0; /* Expected "0 20px 0" to be "0 20px" */
}

```

https://stylelint.io/demo/#N4Igxg9gJgpiBcCBOEIBcAEwA6A7baAtIQA4CGUUAlrgObwYBMADCQB4DceAvnngAIBbGNTIYAzmCQwYuLHzTwU6efjQFi5SjXpNWbDAFZ2XNb1zm8AOjE41W6nQbMMANzJIAFJoqPaASgxmU0tcKwAjVQIHHWc9diME4J4+MLAotBinIPiDZIs8EAAaEAAzKgAbGAA5MmEEEBg2OpIqqzBxcWLwCFxy2ga7DAxsRrY0WShxUYYAbTxh4dHxNABPKoqaIkg+qlpCFbJcKA8oUYWMAF0ebp3+gDEIJEEyNAaAK3Fe7tgSLsQhiMQCt1jBNrh1AggRVXjAVqMihdlmsNltCHc9gc0EcTkgzlDRjCJvCQDduEA

**Comment by Mouvedia** [comment]
> `--padding: 20px 5px;`

To me that's a blocker unless we only inspect the current file.
i.e. if the custom property can be reset the rule shouldn't report

**Comment by romainmenke** [comment]
> this issue should be closed as by design

In that case we should probably make that explicit behavior by adding a test case for it.

I agree that whenever `var()` is involved it becomes exceedingly hard to give correct and meaningful feedback, let alone do auto fixes.

**Comment by ybiquitous** [comment]
Thanks for the feedback. I missed cases like `--padding: 20px 5px;`. I agree `var` should be ignored for this rule. 👍🏼 

I will continue to work on PR #7657 to exclude `var` support since `calc` support still makes sense.

I'm closing this issue.
